daemon/src/wallet.rs

```diff
@@ -109,6 +117,26 @@ impl Actor<ElectrumBlockchain, sled::Tree> {
         let client = bdk::electrum_client::Client::new(electrum_rpc_url)
             .context("Failed to initialize Electrum RPC client")?;
 
+        let network_hash = constants::genesis_block(ext_priv_key.network).block_hash();
+        // This needs to be converted to a BlockHash first, as Sha256d hashes are displayed backward
+        let rpc_hash = BlockHash::from_byte_iter(
+            client
+                .server_features()?
+                .genesis_hash
+                .iter()
+                .map(|c| Ok(*c)),
+        )
+        .context("Invalid genesis block hash returned by electrum RPC")?;
+
+        if rpc_hash != network_hash {
+            tracing::error!(
+                %network_hash,
+                %rpc_hash,
+                "Electrum RPC and network genesis hashes do not match (wrong RPC url for network?)",
+            );
+            return Err(anyhow::anyhow!(WrongRpcNetwork));
+        }
```

Comment:
🔧 Given where `wallet::Actor::spawn` is called (in `main.rs` for `taker` and `maker`), we should just return an error. The log is redundant, because we'll exit with a similar error message anyway!

 I would opt for something like:

```suggestion
        ensure!(rpc_hash == seed_hash, "Electrum RPC and network genesis hashes do not match");
```

Reply:
The log currently prints a bit more info than the error, giving also the hashes themselves. Maybe this is not worth printing, though?

Reply:
Maybe actually we should just log something more obvious and user-friendly like:

```
Wallet seed and Electrum RPC client on different networks.
```

A user should have an easier time figuring out why their application doesn't work then.

Reply:
Done!